### PR TITLE
feat: infer correct Go module path for Sage module

### PR DESCRIPTION
**Pull request by odsod**
Having the correct module prefix is important for exclusion from
license checking.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -126,3 +130,28 @@ func addToDependabot() error {
...
 	return os.WriteFile(dependabotYMLPath, appendSageDependabotConfig(dependabotYML), 0o600)
 }
+
+func resolveSageModulePath(ctx context.Context) (string, error) {
+	const moduleName = ".sage"
+	if _, err := os.Stat("go.mod"); err != nil {
+		return moduleName, nil
+	}
+	var out bytes.Buffer
+	cmd := sg.Command(ctx, "go", "mod", "edit", "-json")
```
Comment (alethenorio) [inline main.go:140]: Can we avoid calling go directly and use the official package for handling gomod files?

https://pkg.go.dev/golang.org/x/mod@v0.5.1/modfile

Comment (odsod) [reply main.go:140]: Since it's in `x/mod` I'd rather not add it as a dependency

Comment (alethenorio) [reply main.go:140]: I think that should be an OK exception since the `golang.org/x` is an official repository from the Go developers. WDYT?

Comment (odsod) [reply main.go]: The point is to not have any external dependencies at all in core Sage, so I don't think the problem is who publishes the module, but to keep the modfile looking like this:

```
module go.einride.tech/sage

go 1.17
```

Comment (alethenorio) [reply main.go]: That's a nice goal but not something we strive prevent at all costs if the alternative is better but we can take that discussion outside this PR

File: main.go [hunk 2]
```diff
@@ -126,3 +130,28 @@ func addToDependabot() error {
...
+			Path string
+		}
+	}
+	if err := json.Unmarshal(out.Bytes(), &modFile); err != nil {
+		return "", err
+	}
+	if modFile.Module.Path == "" {
+		return moduleName, nil
+	}
+	return modFile.Module.Path + "/" + moduleName, nil
```
Comment (alethenorio) [inline main.go:156]:
```suggestion
	return filepath.Join(modFile.Module.Path, moduleName), nil
```

**Review by alethenorio — APPROVED** [review]

**Pull request merged**